src/main/java/org/terasology/logic/players/DefaultMinimapSystem.java

```diff
@@ -125,4 +137,34 @@ public void onPlaceBlock(PlaceBlocks event, EntityRef entity) {
             minimapHUDElement.updateLocation(pos);
         }
     }
+
+    @ReceiveEvent //Spawn, initial spawn on joining a server
+    public void onPlayerSpawnEvent(OnPlayerSpawnedEvent event, EntityRef player) {
+        if (minimapHUDElement != null) {
```

Comment:
Is this `null` check really necessary? I guess due it being a system we cannot make the member `final` and initialize it in the constructor... 
Can we live with the risk of it being `null`?

Reply:
Sometimes the minimapHUDElement is not initialized and players are already getting spawned This mainly happens when a client joins the game, so the spawn event for other players already in the game is called.

Reply:
what happens with the event in that case? will the respective player be added to the minimap in that case?